cime_config/buildnml

```diff
@@ -121,6 +121,16 @@ def _create_namelists(case, confdir, infile, nmlgen):
         if nmlgen.get_value('ice_ic') == 'UNSET':
             nmlgen.set_value('ice_ic', value="default")
 
+    # patch maps are not positive definite, aavg and blin are
+    if case.get_value("ATM2OCN_FMAPNAME").find('patc') == -1:
```

Comment:
Really? It returns -1 if true?

Reply:
Oh wait, the logic is the reverse of what I was thinking. I would rather have the default to always be .true. and only change it when 'patc' is found. Doesn't namelist_definition_cice.xml set these to .true. by default?

Reply:
In `namelist_definition_cice.xml` I defined the variables with `skip_default_entry="true"` because I wanted to use python for the logic of "is this map a patch map or not?". So `buildnml` needs to set the variable to `.false.` for patch maps and `.true.` otherwise -- in python, `str.find(substr)` returns the [0-based] index of the first character matching `substr` or returns -1 if the `substr` is not found. So `case.get_value("ATM2OCN_FMAPNAME").find('patc') == -1` is true if the file name does not contain `patc`.

Reply:
It's maybe a little confusing that the fortran variables track whether the map is positive [semi-]definite or not, given that the interesting case is when the map is NOT positive definite... but I didn't want to name the variables `atm2ice_smap_is_patch` because what happens if another non-positive definite map gets introduced? Either the variable name is misleading or it needs to be changed. On the other hand, `atm2ice_smap_is_not_pos_def` seems awfully clunky.